include/arch/arm/arch/machine/gic_v3.h

```diff
@@ -288,3 +344,195 @@ static inline void ackInterrupt(irq_t irq)
 
 }
 
+#ifdef CONFIG_ARM_HYPERVISOR_SUPPORT
+
+extern unsigned int gic_vcpu_num_list_regs;
+
+static inline uint32_t get_gic_vcpu_ctrl_hcr(void)
+{
+    uint32_t reg;
+    MRS(ICH_HCR_EL2, reg);
+    return reg;
+}
+
+static inline void set_gic_vcpu_ctrl_hcr(uint32_t hcr)
+{
+    MSR(ICH_HCR_EL2, hcr);
+}
+
+static inline uint32_t get_gic_vcpu_ctrl_vmcr(void)
+{
+    uint32_t reg;
+    MRS(ICH_VMCR_EL2, reg);
+    return reg;
+}
+
+static inline void set_gic_vcpu_ctrl_vmcr(uint32_t vmcr)
+{
+    MSR(ICH_VMCR_EL2, vmcr);
+}
+
+/* Note: On the GICv3 there are potentially up to 128 preemption
+ * levels, and as such up to 4 APR registers.
+ *
+ * At this point the code assumes a maximum of 32 preemption levels.
+ */
+static inline uint32_t get_gic_vcpu_ctrl_apr(void)
+{
+    uint32_t reg;
+    MRS(ICH_AP0R0_EL2, reg);
+    return reg;
+}
+
+static inline void set_gic_vcpu_ctrl_apr(uint32_t apr)
+{
+    MSR(ICH_AP0R0_EL2, apr);
+}
+
+static inline uint32_t get_gic_vcpu_ctrl_vtr(void)
+{
+    uint32_t reg;
+    MRS(ICH_VTR_EL2, reg);
+    return reg;
+}
+
+static inline uint32_t get_gic_vcpu_ctrl_eisr0(void)
+{
+    uint32_t reg;
+    MRS(ICH_EISR_EL2, reg);
+    return reg;
+}
+
+/* Note: GICv3 supports a maximum of 16 list registers
+ * so there is no need to support EISR1 on GICv3.
+ */
+static inline uint32_t get_gic_vcpu_ctrl_eisr1(void)
+{
+    return 0;
+}
+
+static inline uint32_t get_gic_vcpu_ctrl_misr(void)
+{
+    uint32_t reg;
+    MRS(ICH_MISR_EL2, reg);
+    return reg;
+}
+
+static inline virq_t get_gic_vcpu_ctrl_lr(int num)
```

Comment:
The other PR addresses this

Reply:
The GIC_VCPU_MAX_NUM_LR is overridden by the global gic_vcpu_num_list_regs which is initialized during boot to the actual number that the hardware supports.  So this would still work if GIC_VCPU_MAX_NUM_LR is set to 64 as that would just impact the size of the struct in the VCPU object.